src/frontend/screens/ManualGpsScreen/index.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from "react";
+import {
+  View,
+  ScrollView,
+  StyleSheet,
+  ToastAndroid,
+  KeyboardAvoidingView,
+} from "react-native";
+import Text from "../../sharedComponents/Text";
+import { FormattedMessage, defineMessages, useIntl } from "react-intl";
+
+import { StackScreenComponent } from "../../types";
+import useSettingsValue from "../../hooks/useSettingsValue";
+import createPersistedState from "../../hooks/usePersistedState";
+import { BLACK } from "../../lib/styles";
+import HeaderTitle from "../../sharedComponents/HeaderTitle";
+import useDraftObservation from "../../hooks/useDraftObservation";
+import LocationContext from "../../context/LocationContext";
+import IconButton from "../../sharedComponents/IconButton";
+import { BackIcon, SaveIcon } from "../../sharedComponents/icons";
+import Select from "../../sharedComponents/Select";
+
+import DdForm from "./DdForm";
+import DmsForm from "./DmsForm";
+import UtmForm from "./UtmForm";
+
+type ConvertedData = {
+  error?: Error;
+  coords?: { lat?: number | null; lon?: number | null };
+};
+
+const m = defineMessages({
+  title: {
+    id: "screens.ManualGpsScreen.title",
+    defaultMessage: "Enter coordinates",
+    description: "title of manual GPS screen",
+  },
+  coordinateFormat: {
+    id: "screens.ManualGpsScreen.coordinateFormat",
+    defaultMessage: "Coordinate Format",
+  },
+  decimalDegrees: {
+    id: "screens.ManualGpsScreen.decimalDegrees",
+    defaultMessage: "Decimal Degrees (DD)",
+  },
+  degreesMinutesSeconds: {
+    id: "screens.ManualGpsScreen.degreesMinutesSeconds",
+    defaultMessage: "Degrees/Minutes/Seconds (DMS)",
+  },
+  universalTransverseMercator: {
+    id: "screens.ManualGpsScreen.universalTransverseMercator",
+    defaultMessage: "Universal Transverse Mercator (UTM)",
+  },
+});
+
+const usePersistedState = createPersistedState("manualCoordinateEntryFormat");
+
+const ManualGpsScreen: StackScreenComponent = ({ navigation }) => {
```

Comment:
the `StackScreenComponent` annotation here is very important for type-checking and inference. helps with props but also the `navigationOptions` configuration on the component level